File: src/main/java/model/equipment/weapon/Halberd.java
package model.equipment.weapon;

import model.support.allEnums.DmgType;
import model.support.allEnums.WeaponType;

public class Halberd extends Weapon {
    public Halberd(double weight, int quality) {
        super(WeaponType.HALBERD, weight, quality);
        setDamage();
    }

    private void setDamage() {
        damage.put(DmgType.BLUNT, quality * 0.05);
        damage.put(DmgType.CUT, quality * 0.05);
        damage.put(DmgType.PIERCING, quality * 0.35);
        damage.put(DmgType.SLASH, quality * 0.55);
    }
}

File: src/main/java/model/equipment/weapon/Halberd.kt
package model.equipment.weapon

import model.support.enums.Skill.HALBERD_SKILL
import model.support.enums.Specification.*
import model.support.enums.WeaponType.HALBERD

class Halberd(weight: Double, quality: Int) : Weapon(HALBERD, HALBERD_SKILL, weight, quality, false) {
    init {
        damage = mapOf(
                BLUNT to quality * 0.05,
                CUT to quality * 0.05,
                SLASH to quality * 0.55,
                PIERCING to quality * 0.35,
                CRUSH to 0.0)
    }
}
